include/ocpp/v201/charge_point.hpp

```diff
@@ -567,6 +579,27 @@ class ChargePoint : public ChargePointInterface, private ocpp::ChargingStationBa
     ///
     void set_evse_connectors_unavailable(EvseInterface& evse, bool persist);
 
+    ///
+    /// \brief Get connector type of Connector
+    /// \param evse_id          EVSE id
+    /// \param connector_id     Connector id
+    /// \return The connector type. If evse or connector id is not correct: std::nullopt.
+    ///
+    std::optional<ConnectorEnum> get_evse_connector_type(const uint32_t evse_id, const uint32_t connector_id);
+
+    ///
+    /// \brief Get connector status.
+    ///
+    /// This will search if there is a connector on this evse with status 'Available'. It will search through all
+    /// connectors and return on the first connector that is 'Available'.
+    ///
+    /// \param evse_id          The evse id.
+    /// \param connector_type   The connector type.
+    /// \return std::nullopt if connector type is given, but does not exist. Otherwise the connector status.
+    ///
+    std::optional<ConnectorStatusEnum> get_available_connector_or_status(const uint32_t evse_id,
```

Comment:
Purely looking at this function from the header it is a bit confusing. I think you are just checking if an evse has a connector available and optionally only check for a specific connector type?

If so, I would rename and change the documentation a bit. The connector_type argument is then a sort of optional filter.

Reply:
Ok I changed it a bit, can you take a look if that's better (have to commit still).

Reply:
This also looks a little bit confusing to me. Looking at how this function is used internally, it looks like `bool is_connector_available(const int32_t evse_id, std::optional<ConnectorEnum> connector_type)` provides everything you need. This would search for an available connector at the given `evse_id` using the given `connector_type` filter. Apart from that it looks like we need a function `bool does_connector_exist(const int32_t evse_id, const ConnectorEnum connector_type)` to check if the given `connector_type` belongs to any evse. I will mark the places in `charge_point.cpp` where I think these function shall be used.

Reply:
Ok, can you (@marcemmers , @Pietfried ) take a look if this is what you mean and if it is more readable now?

Reply:
Maybe `get_connector_status` can be a bit more explicit when it does not find an available connector. Instead of returning the state of one of the connectors I would return the first or the last or at least a defined connector status.

Reply:
It does return the status of the last found connector with the given connector type. But if no connector with that type is found, it returns a nullopt. Should that be changed? Because then there is a status returned of a connector that does not even exist?

I changed the function a bit by the way, if the connector type is unknown, it is also included in the search.

Reply:
Ah sorry, I meant this part `If is no 'Available' connector, it will return the status of one of the connectors.` which makes it seem like it just picks a connector to return the status for.